### ISSUE TITLE
Improved connection handling

### ISSUE THREAD

**Issue by Andrey9kin** [Praqma/tracey-broker#35]
According to Rabbit MQ best practices, we should try to minimize the number of connections (class Connection represents TCP connection to RabbitMQ server) and use a separate channel for every separate thread.

Currently, this part is implemented by RabbitMQConnection class where we will create new connection every time we request new channel

```
    public Channel createChannel() throws IOException, TimeoutException {
        if(channel != null) {
            return channel;
        }
        channel = factory.newConnection().createChannel();
        return channel;
    }
```

So the idea is to extend RabbitMQConnection class with a class variable (static variable) connection that will hold only one connection within an application. And modify createChannel function to create channels from existing connection.

**Comment by Andrey9kin** [comment]
@alexsedova let's get this one fixed first. Then we will do a new release of broker and with that will attempt to solve Praqma/tracey-jenkins-trigger-plugin#55

**Comment by alexsedova** [comment]
@Andrey9kin Why do we need to create a channel for every job?

**Comment by Andrey9kin** [comment]
@alexsedova yes, that is an idea. According to RabbitMQ java, API documentation connections are thread-safe but channels are not so it is better to avoid using one channel for all queues.

**Comment by Andrey9kin** [comment]
@alexsedova so what we can do in the plugin is to have a map that holds channel id instead of consumer tag and then we will close channels via channel id - that will also terminate queues. Which means that we probably need to add one more function to RabbitMQConnection to close channel via channel id

**Comment by alexsedova** [comment]
@Andrey9kin Ok then let's do it)
